common/src/address.rs

```diff
@@ -47,6 +47,68 @@ pub const NEXUS_INTERNAL_PORT: u16 = 12221;
 
 pub const NTP_PORT: u16 = 123;
 
+// The number of ports available to an SNAT IP.
+// Note that for static NAT, this value isn't used, and all ports are available.
+//
+// NOTE: This must be a power of 2. We're expecting to provide the Tofino with a
+// port mask, e.g., a 16-bit mask such as `0b01...`, where those dots are any 14
+// bits. This signifies the port range `[16384, 32768)`. Such a port mask only
+// works when the port-ranges are limited to powers of 2, not arbitrary ranges.
+//
+// Also NOTE: This is not going to work if we modify this value across different
+// versions of Nexus. Currently, we're considering a port range free simply by
+// checking if the _first_ address in a range is free. However, we'll need to
+// instead to check if a candidate port range has any overlap with an existing
+// port range, which is more complicated. That's deferred until we actually have
+// that situation (which may be as soon as allocating ephemeral IPs).
+pub const NUM_SOURCE_NAT_PORTS: u16 = 1 << 14;
+
+lazy_static::lazy_static! {
+    /// The IPv6 prefix assigned to the built-in services VPC.
+    pub static ref SERVICE_VPC_IPV6_PREFIX: Ipv6Net = Ipv6Net(
+        Ipv6Network::new(
+            Ipv6Addr::new(0xfd77, 0xe9d2, 0x9cd9, 0, 0, 0, 0, 0),
+            Ipv6Net::VPC_IPV6_PREFIX_LENGTH,
+        ).unwrap(),
+    );
+
+    /// The IPv4 subnet for External DNS OPTE portsv
+    pub static ref DNS_OPTE_IPV4_SUBNET: Ipv4Net =
```

Comment:
I had these in common because I also wanted to use them while actually creating the VPC & subnet DB records in a follow up change.

Reply:
I wonder if it would make sense to have a crate that both RSS and Nexus use, since it seems like those two will share some infrastructure that we don't need everywhere else.  I wouldn't consider this a blocker here, more like something to consider as we keep working on these things.